test/code-build-test.js

```diff
@@ -84,6 +93,44 @@ describe("githubInputs", () => {
       .to.haveOwnProperty("envPassthrough")
       .and.to.deep.equal(["one", "two", "three", "four"]);
   });
+
+  it("can handle pull requests", () => {
+    // This is how GITHUB injects its input values.
+    // It would be nice if there was an easy way to test this...
+    process.env[`INPUT_PROJECT-NAME`] = projectName;
+    process.env[`GITHUB_REPOSITORY`] = repoInfo;
+    process.env[`GITHUB_SHA`] = sha;
+    process.env[`GITHUB_EVENT_NAME`] = "pull_request";
+    const { context } = require("@actions/github");
+    context.payload = { pull_request: { head: { sha: pullRequestSha } } };
+    const test = githubInputs();
+    expect(test).to.haveOwnProperty("projectName").and.to.equal(projectName);
+    expect(test)
+      .to.haveOwnProperty("sourceVersion")
+      .and.to.equal(pullRequestSha);
+    expect(test).to.haveOwnProperty("owner").and.to.equal(`owner`);
+    expect(test).to.haveOwnProperty("repo").and.to.equal(`repo`);
+    expect(test)
+      .to.haveOwnProperty("buildspecOverride")
+      .and.to.equal(undefined);
+    expect(test).to.haveOwnProperty("envPassthrough").and.to.deep.equal([]);
+  });
+
+  it("will not continue if there is no payload", () => {
+    // This is how GITHUB injects its input values.
+    // It would be nice if there was an easy way to test this...
+    process.env[`INPUT_PROJECT-NAME`] = projectName;
+    process.env[`GITHUB_REPOSITORY`] = repoInfo;
+    process.env[`GITHUB_SHA`] = sha;
+    process.env[`GITHUB_EVENT_NAME`] = "pull_request";
+    // These tests run in pull requests
+    // so to tests things that are NOT pull request...
```

Comment:
not sure, I'm assuming this test doesn't use the environment variable setting at all though right? Since it just wants to test the Error handling

Like this whole test case should just read like:

```javascript
it("will not continue if there is no payload", () => {
	require("@actions/github").context.payload = {};
	expect(() => githubInputs()).to.throw(
		"No source version could be evaluated."
	);
});
```

or are you referring to something else? Pardon my ignorance on this

Reply:
No, this is what I mean :)
The problem is that these test run in a PR...
Because the tests are run in a github action in response to a PR.
BUT, the problem is that that is NOT the only use case for this module.

So to test this if there is no payload,
e.g. you are running this locally or NOT in a PR
I clobber the value so the test can run in the right branch.

This is clearly a case of the test understanding the implementation,
which is generally a bad idea.
I don't have any better ideas...

Reply:
Got it. Would it be reasonable to change GITHUB_EVENT_NAME to something else, or not set it at all? I just suspected that comment didn't actually apply, when technically it does but it's confusing because you have "this is a pull request" in the line right above it. :)

Reply:
No, `GITHUB_EVENT_NAME` is set by GitHub
and it is set to `'pull_request'` when the tests are run...